src/main/java/flab/gotable/service/ReservationService.java

```diff
@@ -0,0 +1,71 @@
+package flab.gotable.service;
+
+import flab.gotable.domain.entity.Reservation;
+import flab.gotable.dto.request.ReservationRequestDto;
+import flab.gotable.dto.response.ReservationResponseDto;
+import flab.gotable.exception.DuplicatedReservationException;
+import flab.gotable.exception.ErrorCode;
+import flab.gotable.exception.ReservationFailedException;
+import flab.gotable.exception.ScheduleNotFoundException;
+import flab.gotable.mapper.ReservationMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.Date;
+
+@Service
+@RequiredArgsConstructor
+public class ReservationService {
+    private final ReservationMapper reservationMapper;
+
+    @Transactional(isolation = Isolation.READ_COMMITTED)
+    public ReservationResponseDto reserve(ReservationRequestDto reservationRequestDto) {
+        final Long restaurantId = reservationRequestDto.getRestaurantId();
+        final LocalDateTime reservationStartTime = reservationRequestDto.getReservationStartTime();
+        final LocalDateTime reservationEndTime = reservationRequestDto.getReservationEndTime();
+
+        // 락 획득 여부
+        if(reservationMapper.getRestaurantLock(reservationRequestDto.getRestaurantId()) > 0) {
+            // 예약하려는 시간에 다른 사용자가 예약하지 않았는지
+            if (!isReservationAvailable(restaurantId, reservationStartTime, reservationEndTime)) {
```

Comment:
ealry throw 패턴을 사용하면 락 획득에 실패한 경우를 좀 더 깔끔하게 처리할 수 있습니다.

```suggestion
        // 락 획득 여부
        if(reservationMapper.getRestaurantLock(reservationRequestDto.getRestaurantId()) <= 0) {
          // 락 획득에 실패
          throw new ...
        }
        
        // 예약하려는 시간이 일반 또는 영업 스케줄 중에 존재하는지
            if (!isExistSchedule(restaurantId, reservationStartTime, reservationEndTime)) {
                throw new ScheduleNotFoundException(ErrorCode.RESERVATION_TIME_NOT_FOUND, ErrorCode.RESERVATION_TIME_NOT_FOUND.getMessage());
            }
            ...
```